pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java

```diff
@@ -178,7 +185,45 @@ public BrokerResponseNative reduceOnDataTable(BrokerRequest brokerRequest,
     }
 
     DataTableReducer dataTableReducer = ResultReducerFactory.getResultReducer(brokerRequest);
-    dataTableReducer.reduceAndSetResults(tableName, cachedDataSchema, dataTableMap, brokerResponseNative, brokerMetrics);
+    dataTableReducer
+        .reduceAndSetResults(tableName, cachedDataSchema, dataTableMap, brokerResponseNative, brokerMetrics);
+    updateAliasToSchemaName(brokerRequest, brokerResponseNative);
     return brokerResponseNative;
   }
-}
+
+  private static void updateAliasToSchemaName(BrokerRequest brokerRequest, BrokerResponseNative brokerResponseNative) {
+    if (brokerRequest.getPinotQuery() == null) {
+      return;
+    }
+    QueryOptions queryOptions = new QueryOptions(brokerRequest.getQueryOptions());
+    if (!queryOptions.isResponseFormatSQL()) {
+      return;
+    }
+    DataSchema dataSchema = brokerResponseNative.getResultTable().getDataSchema();
+    List<Expression> selectList = brokerRequest.getPinotQuery().getSelectList();
+    String[] columnNames = dataSchema.getColumnNames();
+    if (columnNames.length != selectList.size()) {
+      LOGGER.error("Response schema size {} is different from number of selection expressions {}", columnNames.length,
+          selectList.size());
+      return;
+    }
+    for (int i = 0; i < selectList.size(); i++) {
+      if (selectList.get(i).getFunctionCall() != null && selectList.get(i).getFunctionCall().getOperator()
+          .equalsIgnoreCase("AS")) {
+        String aliasName = null;
+        if (selectList.get(i).getFunctionCall().getOperands().get(1).getIdentifier() != null) {
+          aliasName = selectList.get(i).getFunctionCall().getOperands().get(1).getIdentifier().getName();
+        }
+        Literal aliasLiteral = selectList.get(i).getFunctionCall().getOperands().get(1).getLiteral();
+        if (aliasLiteral != null) {
+          if (aliasLiteral.isSetStringValue()) {
+            aliasName = aliasLiteral.getStringValue();
+          }
+        }
+        if (aliasName != null) {
+          columnNames[i] = aliasName;
+        }
```

Comment:
Can you simplify the logic here and add some comments? Can `aliasName` be `null` when `AS` keyword is found?

Reply:
technically it shouldn't be null, this is just for sanity check.